_source/_posts/2023-07-27-spring-graphql-react.md

```diff
@@ -0,0 +1,1794 @@
+            newObject.id = id;
+            return newObject;
+        }
+    }
+
+    public String getSIC() {
+        return SIC;
+    }
+
+    public void setSIC(String SIC) {
+        this.SIC = SIC;
+    }
+
+    public String getCategory() {
+        return category;
+    }
+
+    public void setCategory(String category) {
+        this.category = category;
+    }
+
+    public String getCompanyNumber() {
+        return companyNumber;
+    }
+
+    public void setCompanyNumber(String companyNumber) {
+        this.companyNumber = companyNumber;
+    }
+
+    public String getCountryOfOrigin() {
+        return countryOfOrigin;
+    }
+
+    public void setCountryOfOrigin(String countryOfOrigin) {
+        this.countryOfOrigin = countryOfOrigin;
+    }
+
+    public LocalDate getIncorporationDate() {
+        return incorporationDate;
+    }
+
+    public void setIncorporationDate(LocalDate incorporationDate) {
+        this.incorporationDate = incorporationDate;
+    }
+
+    public Integer getMortgagesOutstanding() {
+        return mortgagesOutstanding;
+    }
+
+    public void setMortgagesOutstanding(Integer mortgagesOutstanding) {
+        this.mortgagesOutstanding = mortgagesOutstanding;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public String getStatus() {
+        return status;
+    }
+
+    public void setStatus(String status) {
+        this.status = status;
+    }
+}
+```
+
+Create the package `com.okta.developer.demo.repository` and the class `CompanyRepository`:
+
+__CompanyRepository.java__
+```java
+package com.okta.developer.demo.repository;
+
+import com.okta.developer.demo.domain.Company;
+import org.springframework.data.neo4j.repository.ReactiveNeo4jRepository;
+
+public interface CompanyRepository extends ReactiveNeo4jRepository<Company, Long> {
+
+}
+```
+
+Create the configuration class `GraphQLConfig` under the root package. This class will enable CORS from the React client and log the GraphQL schema mappings:
+
+__GraphQLConfig.java__
+```java
+package com.okta.developer.demo;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.autoconfigure.graphql.GraphQlSourceBuilderCustomizer;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration(proxyBeanMethods = false)
+class GraphQLConfig {
+
+    private static Logger logger = LoggerFactory.getLogger("graphql");
+
+    @Bean
+    public GraphQlSourceBuilderCustomizer sourceBuilderCustomizer() {
+        return (builder) ->
+                builder.inspectSchemaMappings(report -> {
+                    logger.debug(report.toString());
+                });
+    }
+}
+```
+
+Create the configuration class `SpringBootGraphQlApiConfig` in the root package as well, defining a reactive transaction manager required for reactive Neo4j:
```

Comment:
```suggestion
Create a configuration class named `SpringBootGraphQlApiConfig` in the root package as well, defining a reactive transaction manager required for reactive Neo4j:
```

Reply:
I think we should be consistent as to how you capitalize GraphQL. In the previous class, you have an uppercase L and here you have a lowercase L.

Reply:
I did it t match the application class created by spring initializr. Maybe I should change the application name too?